lib/cubit/webhook/lnurl_pay_service.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:breez_sdk_liquid/breez_sdk_liquid.dart';
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+import 'package:http/http.dart' as http;
+import 'package:l_breez/cubit/webhook/webhook_state.dart';
+import 'package:logging/logging.dart';
+
+final Logger _logger = Logger('LnUrlPayService');
+
+class LnUrlPayService {
+  static const String lnurlServiceURL = 'https://breez.fun';
+
+  final BreezSDKLiquid _breezSdkLiquid;
+  final BreezPreferences _breezPreferences;
+
+  LnUrlPayService(this._breezSdkLiquid, this._breezPreferences);
+
+  Future<Map<String, String>> registerLnurlpay(
+    WalletInfo walletInfo,
+    String webhookUrl, {
+    String? username,
+  }) async {
+    final String pubKey = walletInfo.pubkey;
+
+    await _invalidatePreviousWebhookIfNeeded(pubKey, webhookUrl);
+
+    final int currentTime = DateTime.now().millisecondsSinceEpoch ~/ 1000;
+
+    // TODO(erdemyerebasmaz): Utilize user's username(only when user has created a new wallet)
+    // TODO(erdemyerebasmaz): Handle multiple device setup cases
+    final String? lnAddressUsername =
+        (username ?? await _breezPreferences.getProfileName())?.replaceAll(' ', '');
+
+    final SignMessageResponse? signMessageRes = await _generateSignature(
+      webhookUrl: webhookUrl,
+      currentTime: currentTime,
+      lnAddressUsername: lnAddressUsername,
+    );
+
+    if (signMessageRes == null) {
+      throw Exception('Missing signature');
+    }
+
+    final Map<String, String> response = await _postWebhookRegistrationRequest(
+      pubKey,
+      webhookUrl,
+      lnAddressUsername,
+      signMessageRes.signature,
+    );
+
+    await setLnUrlPayKey(webhookUrl: webhookUrl);
+    return response;
+  }
+
+  Future<void> _invalidatePreviousWebhookIfNeeded(String pubKey, String webhookUrl) async {
+    final String? lastUsedLnurlPay = await getLnUrlPayKey();
+    if (lastUsedLnurlPay != null && lastUsedLnurlPay != webhookUrl) {
+      await _invalidateLnurlPay(pubKey, lastUsedLnurlPay);
+    }
+  }
+
+  Future<SignMessageResponse?> _generateSignature({
+    required String webhookUrl,
+    required int currentTime,
+    String? lnAddressUsername,
+  }) async {
+    final String username = lnAddressUsername?.isNotEmpty == true ? '-$lnAddressUsername' : '';
+    final String message = '$currentTime-$webhookUrl$username';
+
+    final SignMessageRequest req = SignMessageRequest(message: message);
+    return _breezSdkLiquid.instance?.signMessage(req: req);
+  }
+
+  Future<Map<String, String>> _postWebhookRegistrationRequest(
+    String pubkey,
+    String webhookUrl,
+    String? username,
+    String signature,
+  ) async {
+    final String lnurlWebhookUrl = '$lnurlServiceURL/lnurlpay/$pubkey';
+    final Uri uri = Uri.parse(lnurlWebhookUrl);
+
+    final http.Response jsonResponse = await http.post(
+      uri,
+      body: jsonEncode(
+        AddWebhookRequest(
+          time: DateTime.now().millisecondsSinceEpoch ~/ 1000,
```

Comment:
The `time` needs to be the same as the one signed in the message

Reply:
Missed that one. Thanks.